core/object/object.cpp

```diff
@@ -262,6 +262,9 @@ bool Object::_predelete() {
 }
 
 void Object::cancel_free() {
+	if (_predelete_ok) {
+		_is_queued_for_deletion = false;
+	}
```

Comment:
What would be the problem with always setting `_is_queued_for_deletion` to `false`, regardless of `_predelete_ok`?